### PR TITLE
Implement fixed sized byte arrays (closes #239)

### LINKED ISSUE #239: Audit the byte array types and annotations.
https://github.com/ethereum/eth1.0-specs/pull/219

### PR DESCRIPTION
**Pull request by SamWilsn**
(pinging @Bhargavasomu , @ultratwo & @voith for reviews please!)

### What was wrong?

Related to Issue #239 

### How was it fixed?

Created a fixed sized byte array class and subclasses for specific sizes.

#### Cute Animal Picture
photo_2021-09-23_17-01-15-crop

### CODE REVIEW

**Review by ultratwo — APPROVED** [review]
LGTM!

**Review by Bhargavasomu — APPROVED** [review]
Looks good.

File: src/ethereum/frontier/trie.py [hunk 1]
```diff
@@ -49,13 +49,16 @@
...
+EMPTY_TRIE_ROOT = Root(
+    hex_to_bytes(
+        "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"
+    )
 )
 
 Node = Union[Account, Bytes, Transaction, Receipt, Uint, U256, None]
-T = TypeVar(
-    "T",
+K = TypeVar("K", bound=Bytes)
```
Comment (voith) [inline src/ethereum/frontier/trie.py:59]: Can we avoid single-letter names like these? Can we call this `TrieKey` instead?

File: src/ethereum/frontier/trie.py [hunk 2]
```diff
@@ -49,13 +49,16 @@
...
+V = TypeVar(
```
Comment (voith) [inline src/ethereum/frontier/trie.py:60]: Can we call this `TrieValue` instead?

File: src/ethereum/frontier/vm/interpreter.py [hunk 3]
```diff
@@ -69,10 +69,14 @@ def process_message_call(
 
     Returns
     -------
-    output : `Tuple[U256, List[eth1spec.eth_types.Log]]`
-        The tuple `(gas_left, logs)`, where `gas_left` is the remaining gas
-        after execution, and logs is the list of `eth1spec.eth_types.Log`
-        generated during execution.
+    output : `Tuple`
+        A tuple of the following:
```
Comment (voith) [inline src/ethereum/frontier/vm/interpreter.py:73]: Thanks for fixing this.

**Review by voith — APPROVED** [review]
LGTM

File: src/ethereum/frontier/trie.py (hunk 1 above)
Comment (SamWilsn) [reply src/ethereum/frontier/trie.py:59]: Ah, I didn't realize you were talking about generic arguments on the sync call. I fairly strongly prefer single letter generic arguments.

Comment (voith) [reply src/ethereum/frontier/trie.py:59]: Okay, I don't have a strong opinion in that case. Feel free to merge.

**Pull request merged**